wikimedia/wikipedia-ios
# Fix editing title descriptions when selected from the context menu of selected text

**Fixes Phabricator ticket:**  https://phabricator.wikimedia.org/T252042

### Notes
* I had to bring in a bit more data from the JS side to figure out how to present the info for title descriptions. This is my first time messing with the app-supplied JS code, so please let me know if I did things in a non-standard way.
* Notably, on the JS side I couldn't find a way to access the PCS-supplied `EditTransform` class from the app-supplied JS code. I've hardcoded the values, which makes it brittle (while showing the source in a comment). Is there a better way fro me to handle this?

### Test Steps
On an article which uses the title description menu, such as `USCGC Dallas (WHEC-716)`:
1. Long press text in the intro section. Select "Edit" from the context menu. Confirm that the edit article VC opens with the first section, and the app doesn't give the "edit intro or first section" menu.
1. Long press text in the second section. Select "Edit" from the context menu. Confirm that the edit article VC opens with the second section, and the app doesn't give the "edit intro or first section" menu.
1. 1. Long press text in the title description. Select "Edit" from the context menu. Confirm that the edit title description VC opens, and the app doesn't give the "edit intro or first section" menu.

On an article which does NOT use the title description menu, such as `Asian Giant Hrnet`:
1. Long press text in the intro section. Select "Edit" from the context menu. Confirm that the edit article VC opens with the first section, and the app doesn't give the "edit intro or first section" menu.
1. Long press text in the second section. Select "Edit" from the context menu. Confirm that the edit article VC opens with the second section, and the app doesn't give the "edit intro or first section" menu.
1. Long press text in the title description. Select "Edit" from the context menu. Confirm that the edit article VC opens with the first section showing and short description selected, and the app doesn't give the "edit intro or first section" menu.
